Svc/TlmChan/TlmChan.cpp

```diff
@@ -62,23 +64,58 @@ void TlmChan::TlmGet_handler(FwIndexType portNum, FwChanIdType id, Fw::Time& tim
     FwChanIdType index = this->doHash(id);
 
     // Search to see if channel has been stored
-    TlmEntry* entryToUse = this->m_tlmEntries[this->m_activeBuffer].slots[index];
+    // check both buffers
+    // don't need to lock because this port is guarded
+    TlmEntry* activeEntry = this->m_tlmEntries[this->m_activeBuffer].slots[index];
     for (FwChanIdType bucket = 0; bucket < TLMCHAN_HASH_BUCKETS; bucket++) {
-        if (entryToUse) {  // If bucket exists, check id
-            if (entryToUse->id == id) {
+        if (activeEntry) {  // If bucket exists, check id
+            if (activeEntry->id == id) {
                 break;
             } else {  // otherwise go to next bucket
-                entryToUse = entryToUse->next;
+                activeEntry = activeEntry->next;
             }
         } else {  // no buckets left to search
             break;
         }
     }
 
-    if (entryToUse) {
-        val = entryToUse->buffer;
-        timeTag = entryToUse->lastUpdate;
-    } else {  // requested entry may not be written yet; empty buffer
+    TlmEntry* inactiveEntry = this->m_tlmEntries[1 - this->m_activeBuffer].slots[index];
+    for (FwChanIdType bucket = 0; bucket < TLMCHAN_HASH_BUCKETS; bucket++) {
+        if (inactiveEntry) {  // If bucket exists, check id
+            if (inactiveEntry->id == id) {
+                break;
+            } else {  // otherwise go to next bucket
+                inactiveEntry = inactiveEntry->next;
+            }
+        } else {  // no buckets left to search
+            break;
+        }
+    }
+
+    if (activeEntry && inactiveEntry) {
+        // two entries. grab the one with the most recent time tag
+        if (Fw::Time::compare(inactiveEntry->lastUpdate, activeEntry->lastUpdate) == Fw::Time::Comparison::GT) {
+            // inactive entry is more recent
+            val = inactiveEntry->buffer;
+            timeTag = inactiveEntry->lastUpdate;
+            return;
+        } else {
+            // active entry is more recent, or they are equal
+            val = activeEntry->buffer;
```

Comment:
In the case of UNCOMPARIBLE, both are valid, but no way to infer which is newer.   Thus one was just picked. 

Could the updated flag be used? If one is `updated` and the other is not, that means it has new data.   If both are updated or not, then choose one.

It is a question, I am not an expert on this code.